### ISSUE TITLE
Latest Modin as a data source for pandas-ai

### ISSUE THREAD

**Issue by YarShev** [Sinaptik-AI/pandas-ai#1001]
### 🚀 The feature

Hey folks, I just saw you had added support for Modin as a data source for pandas-ai in https://github.com/Sinaptik-AI/pandas-ai/pull/907. This is a great addition to the other data sources like pandas and polars, for instance, that has already been introduced much earlier. I am a little concerned about the Modin version pandas-ai currently supports, namely, [0.18.1](https://github.com/Sinaptik-AI/pandas-ai/blob/66dea557956a6e378856472013340c2ced9215c4/poetry.lock#L3369). It is too old and does not contain latest perfomance features and optimizations. Would it be possible to support a latest Modin version?

### Motivation, pitch

All libraries are advancing and making great progress with respect to ultimate performance on latest versions and pandas with Modin are not exceptions. To speed up chatting with data through pandas-ai and reduce users' waiting time it would be great to support latest versions of Modin, pandas, etc.

**Comment by gventuri** [comment]
Thanks a lot for opening the issue, we'll try to update the dependency asap. If anyone wants to contribute, feel free to :)

**Comment by mspronesti** [comment]
@gventuri @YarShev `modin` depends on `pandas`, so unless we upgrade `pandas` we can't do so with modin.

**Comment by mspronesti** [comment]
@YarShev I'm not sure. As far as I know, pandas 2.x is 100% backward compatible with pandas 1.x, so I don't foresee any problems - despite modern LLMs have never "seen" the new methods of the API in their training data. Even langchain uses pandas 2.x ([source](https://github.com/langchain-ai/langchain/blob/master/libs/experimental/pyproject.toml#L21)). However, I've read of people stating the opposite, e.g. this [comment](https://github.com/Sinaptik-AI/pandas-ai/issues/797#issuecomment-1905413469).

**Comment by YarShev** [comment]
@mspronesti, can we give it a try to upgrade pandas and Modin? Are there all necessary tests for testing this?

**Comment by mspronesti** [comment]
@YarShev I'm not an official maintainer of this project, but a "loyal contributor" :-). I let @gventuri respond to your inquiry - I believe any error arising from the upgrade would (possibly) lie in the code generated by the LLMs, which we don't test here as external APIs are mocked.

**Comment by gventuri** [comment]
@YarShev @mspronesti we decided to downgrade to pandas < 2 a while ago as there were some compatibility issues. We could try to upgrade both pandas and modin and release an experimental version for us to try and if no significant issues arise, we can try to fix them.

Given the fact that knowledge cutoff has been updated in the meanwhile, I expect the auto correcting framework to now be in able to fix possible compatibility errors.
